pkg/utils/net/ip.go

```diff
@@ -48,3 +49,16 @@ func ToIPFamily(ipStr string) corev1.IPFamily {
 	}
 	return corev1.IPv6Protocol
 }
+
+// IPLiteralFor returns the given IP as a literal that can be used in a resource identifier.
+// For IPv6 that means returning a bracketed version of the IP.
+// The difference to net.JoinHostPort is that it also allows IP be to be a placeholder that will be resolved
+// to the actual IP at a later time.
+func IPLiteralFor(ipOrPlaceholder string, ipFamily corev1.IPFamily) string {
+	var prefix, suffix string
```

Comment:
nit: I was wondering if we really need those variables since they are used only for IPv6.
Any reason to not avoid them and use something more explicit ?

```golang
func IPLiteralFor(ipOrPlaceholder string, ipFamily corev1.IPFamily) string {
	if ipFamily == corev1.IPv6Protocol {
                 // IPv6: return a bracketed version of the IP
		return fmt.Sprintf("[%s]", ipOrPlaceholder)
	}
        // IPv4: leave the place holder as is
	return ipOrPlaceholder
}
```